src/actions/processes/spoofing.rs

```diff
@@ -37,25 +47,58 @@ pub struct Spoofing {
         help = "Full path to the executable eg: c:\\temp..."
     )]
     executable: String,
+    #[clap(
+        short = 'p',
+        long,
+        required = true,
+        help = "Full path to the parent executable eg: c:\\temp..."
+    )]
+    parent_executable: String,
+}
+
+#[derive(Debug)]
+struct ProcessNotFound;
+
+impl Error for ProcessNotFound {}
+
+impl Display for ProcessNotFound {
+    fn fmt(&self, formatter: &mut Formatter) -> FormatterResult {
+        write!(formatter, "Process not found")
+    }
 }
 
-fn get_user_pid() -> u32 {
-    let s: System = System::new_all();
-    let mut ppid_list: Vec<u32> = Vec::<u32>::new();
-    for (pid, process) in s.processes() {
-        if process.user_id().is_some() {
-            ppid_list.push(pid.as_u32());
+fn get_pid_from_name(name: &str) -> Result<u32, Box<dyn Error>> {
+    let snapshot: Owned<HANDLE> =
+        unsafe { Owned::new(CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0)?) };
+    let mut process_entry: PROCESSENTRY32W = PROCESSENTRY32W::default();
+    process_entry.dwSize = size_of::<PROCESSENTRY32W>() as u32;
+
+    unsafe {
+        Process32FirstW(*snapshot, &mut process_entry)?;
+    }
+
+    loop {
+        if OsString::from_wide(
+            process_entry
+                .szExeFile
+                .into_iter()
+                .take_while(|&byte| byte != 0)
+                .collect::<Vec<_>>()
+                .as_slice(),
+        ) == name
+        {
+            return Ok(process_entry.th32ProcessID);
+        }
+
+        if unsafe { Process32NextW(*snapshot, &mut process_entry) }.is_err() {
+            break;
         }
     }
-    let new_ppid: u32 = ppid_list
-        .choose(&mut rand::thread_rng())
-        .unwrap()
-        .to_owned();
-    new_ppid
+
+    Err(Box::new(ProcessNotFound))
```

Comment:
Can not be use un a CI/CD if we use windows boxes

Reply:
Box is for heap allocation :smile: